### PR TITLE
Split add-on available exceptions

### PR DESCRIPTION
**Pull request by ludeeus**
## Proposed change

Split the exception for why an addon is not available

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (which adds functionality to the supervisor)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [x] Code quality improvements to existing code or addition of tests

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] The code has been formatted using Black (`black --fast supervisor tests`)
- [x] Tests have been added to verify that the new code works.

If API endpoints of add-on configuration are added/changed:

- [ ] Documentation added/updated for [developers.home-assistant.io][docs-repository]

[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[docs-repository]: https://github.com/home-assistant/developers.home-assistant

### CODE REVIEW

**Review by MartinHjelmare — APPROVED** [review]
Nice!

**Review by MartinHjelmare — APPROVED** [review]

File: supervisor/addons/model.py [hunk 1]
```diff
@@ -102,6 +102,16 @@ def is_installed(self) -> bool:
...
+    @property
+    def available_arch(self) -> bool:
+        """Return True if the running arch is supported by the add-on."""
+        return self._available_arch(self.data)
+
+    @property
+    def available_homeassistant(self) -> bool:
+        """Return True if the running version of Home Assistant is supported by the add-on."""
+        return self._available_homeassistant(self.data)
+
```
Comment (pvizeli) [inline supervisor/addons/model.py:114]: Let's put this into AddStore model

Comment (ludeeus) [reply supervisor/addons/model.py:114]: Had to revert it, it's used in restore, so too much would be needed to be moved.

**Comment by pvizeli** [comment]
Now it's a bit nested and mixup with old/new style. Let's doing the following:
```
class HassioError(Exception):
    """Root exception."""

   def __init__(self, mgs: Optional[str] = None, logger: Optional[logging.Logger]):
       ```....```
       if logger:
          logger.error(msg)
       ...
```
That safe us the nested if and we can handle this 2 cases a nesting above, also it will log it to user

**Comment by ludeeus** [comment]
No doing that for this change, the only thing that changed here is that the clients get more info in the exception captured and sent instead of the generic "Unknown error, see supervisor."

**Pull request closed without merging**